temporal/server_option.go

```diff
@@ -96,8 +94,11 @@ func WithClaimMapper(claimMapper func(cfg *config.Config) authorization.ClaimMap
 	})
 }
 
-// Set custom tally metric reporter
-func WithCustomMetricsReporter(reporter tally.BaseStatsReporter) ServerOption {
+// Set custom metric reporter
+// for (deprecated) Tally it should be tally.BaseStatsReporter
+// for Prometheus with framework metrics.FrameworkCustom it should be metrics.Reporter
+// not used otherwise
+func WithCustomMetricsReporter(reporter interface{}) ServerOption {
```

Comment:
Should you add TODO saying to change parameter type to `Reporter` when tally is removed?